fix: radio styling

seaerchin opened the pull request:
Closes #4295

## 📝 Description
`Radio` styling was being applied to the checkbox rather than the container. This behaviour was different from `Checkbox` (the chakra checkbox component), where the styling was applied to the container. 

As the radio is an extremely small element and its colours can be adjusted through usage of the `colorScheme` prop, it makes more sense for the behaviour to be brought in line with `Checkbox`, where styling is applied to the container as a whole.

This was done through making the props being applied to the container take `styleProps` (which is a strict superset of the old `layoutProps`). This still preserves behaviour where additional `htmlProps` is being applied to the checkbox (of the `Radio`)

## ⛳️ Current behavior (updates)

Styling for `Radio` is being applied onto the checkbox element rather than the container element, which is different from what `Checkbox` establishes and is also counter-intuitive. 

## 🚀 New behavior
This PR changes the internals of `Radio` so that `styleProps` are also applied rather than just `layoutProps`

## 💣 Is this a breaking change (Yes/No):
No

## 📝 Additional Information
-

seaerchin commented:
not too sure why the upstream doesn't have these commits but i can rebase and fix if needed

simontaisne reviewed (commented):
Thanks for the PR @seaerchin. The code makes sense to me. Anything missing before we can merge this one @segunadebayo?

seaerchin commented:
Hi @simontaisne, apologies for the absence as i got caught up with schoolwork!! Let me know if there's anything needed! i'll rebase off the current head ref!

simontaisne commented:
No need to apologize, thank you for your work! Lmk when it's good to review 👍

seaerchin commented:
@simontaisne should be good for review!

simontaisne commented on packages/radio/stories/radio.stories.tsx:
```diff
@@ -23,7 +23,17 @@ export const Readonly = () => (
   </Radio>
 )
-
+export const StyledRadio = () => (
```
Could we maybe have a similar story for `Checkbox`?

seaerchin replied:
added! thanks for the suggestion!

simontaisne reviewed (approved):
Looking good! Thanks for the fix @seaerchin 👍

Not sure about the changes on the members & sponsors though?

seaerchin commented:
@simontaisne unsure why that happened with the sponsors actually as i rebased and took the original repo's ref all the way; i rebased and dropped the erroneous commits so it should be good now!

simontaisne commented:
Hi @segunadebayo @TimKolberger @ljosberinn could we please get a review from a maintainer here :)

anubra266 commented:
cc @TimKolberger

segunadebayo commented:
Thanks for working on this @seaerchin.

My sincere apologies for the delayed response. I consider this a breaking change given how long ago this PR/Issue was raised. We'll add it to the next major release.

seaerchin commented:
If it's breaking solely because of age, I'm happy to rebase off the current HEAD ref! Lmk if I need to do anything else beyond that @segunadebayo
